### PR TITLE
📚 Add kdoc for Square.* and Testing., ...

### PR DESCRIPTION
**Pull request by jmfayard**

### CODE REVIEW

**Comment by jmfayard** [comment]
do not merge yet, will continue later

**Comment by jmfayard** [comment]
ready for review

File: plugins/dependencies/src/main/kotlin/dependencies/Kotlin.kt [hunk 1]
```diff
@@ -7,7 +7,19 @@ import org.gradle.kotlin.dsl.IsNotADependency
...
+     * https://kotlinlang.org/api/latest/jvm/stdlib/
+     */
     val stdlib = Stdlib
+
+    /**
+     * the kotlin.test library provides annotations to mark test functions
+     * and a set of utility functions for performing assertions in tests, independently of the test framework being used.
+     *
+     * https://kotlinlang.org/api/latest/kotlin.test/
+     * https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/
```
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Kotlin.kt:21]: These dangling links are not clickable in the IDE. Let's avoid that and always have text setting expectations on what the link is about.

File: plugins/dependencies/src/main/kotlin/dependencies/CashApp.kt [hunk 2]
```diff
@@ -4,6 +4,13 @@ import org.gradle.kotlin.dsl.IsNotADependency
 
 object CashApp {
 
+    /**
+     * SQLDelight - Generates typesafe Kotlin APIs from SQL
+     *
+     * - [Official Website - API and documentation](https://cashapp.github.io/sqldelight/)
+     * - [cashapp/sqldelight: GitHub](https://github.com/cashapp/sqldelight)
+     * - [CHANGELOG](https://github.com/cashapp/sqldelight/blob/master/CHANGELOG.md)
```
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/CashApp.kt:12]: The style here is inconsistent with the existing dependency notations KDoc that can be found in the `Square` object.

Having a consistent style makes a better impression on users and values the effort in making the whole project right.

Happy to improve the style, but we need to discuss it and agree on a style to stick with.
I'll complete a contributing guide covering this as we might welcome further contributions there from the community, and we might forget about it ourselves.

I'm not repeating this comment to all the other KDocs in this PR, this one stands for them all.

File: plugins/dependencies/src/main/kotlin/dependencies/KotlinX.kt [hunk 3]
```diff
@@ -5,16 +5,68 @@ import org.gradle.kotlin.dsl.IsNotADependency
...
+     * - [Kotlin/JS Overview](https://kotlinlang.org/docs/reference/js-overview.html)
+     */
     const val nodeJs = "$artifactBase-nodejs:_"
 
+    /**
+     * Kotlin/kotlinx.reflect.lite - Lightweight library allowing to introspect basic stuff about Kotlin symbols
+     *
+     * - [GitHub](https://github.com/Kotlin/kotlinx.reflect.lite)
+     * - https://kotlinlang.org/docs/reference/reflection.html
+     * - https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.reflect/
```
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/KotlinX.kt:68]: Dangling links

File: plugins/dependencies/src/main/kotlin/dependencies/Orchid.kt [hunk 4]
```diff
@@ -4,13 +4,23 @@ import org.gradle.api.Incubating
...
+ */
 object Orchid {
 
     private const val groupId = "io.github.javaeden.orchid"
 
     const val core = "$groupId:OrchidCore:_"
     const val test = "$groupId:OrchidTest:_"
 
+    /**
+     * https://orchid.run/plugins - Orchid plugins
```
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Orchid.kt:22]: Dangling link

File: plugins/dependencies/src/main/kotlin/dependencies/Orchid.kt [hunk 5]
```diff
@@ -4,13 +4,23 @@ import org.gradle.api.Incubating
 import org.gradle.kotlin.dsl.IsNotADependency
 
 @Incubating
+/**
+ * orchidhq/Orchid - Build and deploy beautiful documentation sites that grow with you
+ * - [GitHub](https://github.com/orchidhq/Orchid)
+ * - [QuickStart & Documentation](https://orchid.run/wiki/user-manual/getting-started/quickstart)
+ * - https://orchid.run/plugins - Orchid plugins
```
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Orchid.kt:11]: Dangling link

File: plugins/dependencies/src/main/kotlin/dependencies/Square.kt [hunk 6]
```diff
@@ -7,10 +7,51 @@ import org.gradle.kotlin.dsl.IsNotADependency
...
+     */
     val okHttp3 = OkHttp3
+
+    /**
+     * Retrofit - A type-safe HTTP client for Android and the JVM
+     *
+     * - [Website](https://square.github.io/retrofit/)
+     * - [GitHub square/retrofit](https://github.com/square/retrofit)
+     * - [CHANGELOG](https://github.com/square/retrofit/blob/master/CHANGELOG.md)
+     * - [Consuming APIs with Retrofit | CodePath Android Cliffnotes](https://guides.codepath.com/android/Consuming-APIs-with-Retrofit#references)
```
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Square.kt:25]: We don't control how outdated a third-party article can become, yet, we are helping our users update (diverging further from these possibly outdated articles at a quicker rate).

Also, we might never find the best article on a library, so let's stick to the official and blog posts from authors and major contributors, and for the rest, or when official content is lacking, let the users decide what to look for as a reference based on the metrics they value like time of publication, programming language (Kotlin vs Java), platform (Android, JVM, other…), search engines and critical sense do a great job for that.

**Review by LouisCAD — CHANGES_REQUESTED** [review]
I've already started correcting the style to be consistent, and addressing other comments I made.

I'll merge this once I'm done with it (will happen before Monday).

**Review by LouisCAD — APPROVED** [review]

**Comment by LouisCAD** [comment]
@jmfayard Feel free to review the changes I just pushed and merge once the CI is done running the checks.

**Pull request merged**